homeassistant/components/onewire/sensor.py

```diff
@@ -426,11 +427,29 @@ def state(self) -> StateType:
         """Return the state of the entity."""
         return self._state
 
-    def update(self):
+    async def get_temperature(self):
+        """Get the latest data from the device."""
+        attempts = 1
+        while True:
+            try:
+                return self._owsensor.get_temperature()
```

Comment:
We're not allowed to do blocking I/O in the event loop. This needs to be scheduled on the executor. Use `hass.async_add_executor_job`.